index.js

```diff
@@ -751,6 +827,32 @@
            keys.every(function(k) { return equals(self[k], other[k]); });
   }
 
+  function unionKeys(a, b) {
+    var keys = {};
+    for (var key in a) keys[key] = 0;
+    for (key in b) keys[key] = 0;
+    return Object.keys(keys);
+  }
+
+  //  Object$prototype$lte :: StrMap a ~> StrMap a -> Boolean
+  function Object$prototype$lte(other) {
+    var keys = unionKeys(this, other).sort();
+    for (var idx = 0; idx < keys.length; idx += 1) {
+      var key = keys[idx];
+      var a = this[key], b = other[key];
+
+      if (!equals(a, b)) {
+        var inThis = Object.prototype.propertyIsEnumerable.call(this, key);
+        var inOther = Object.prototype.propertyIsEnumerable.call(other, key);
+        return  inThis === false ? true :
+                inOther === false ? false :
+                lte(a, b);
+      }
+    }
+
+    return true;
+  }
+
```

Comment:
Looking at [Data.StringMap](http://hackage.haskell.org/package/data-stringmap-1.0.1.1/docs/Data-StringMap.html) would be a good start.